### PR TITLE
Hosted content selective loading

### PR DESCRIPTION
**Pull request by rich-nguyen**
This addresses the following:
- we are currently loading the media js app on every page!
- there is an initialisation race condition between enhanced media, and hosted video media. This means that hosted video often fails to upgrade, and appears unskinned.
- hosted pages fail to run the `showUserName` inline script without throwing the error: `Cannot set property 'innerHTML' of undefined`

### CODE REVIEW

**Comment by kelvin-chappell** [comment]
Hurray! 👍

**Comment by Calanthe** [comment]
👍

File: static/src/javascripts/bootstraps/commercial.js [hunk 1]
```diff
@@ -72,6 +69,13 @@ define([
         }]
     ];
 
+    if (config.isHosted) {
+        secondaryModules.unshift(
```
Comment (regiskuckaertz) [inline static/src/javascripts/bootstraps/commercial.js]: Do we want these module to boot before `dfpLoad`?

**Comment by regiskuckaertz** [comment]
Yeah 👍

File: static/src/javascripts/bootstraps/commercial.js (hunk 1 above)
Comment (rich-nguyen) [inline static/src/javascripts/bootstraps/commercial.js]: It's fine, there aren't any dependencies on `dfpLoad`.

**Pull request merged**